llvm/include/llvm/SYCLLowerIR/SYCLFrameworkOptimization.h

```diff
@@ -0,0 +1,31 @@
+//== SYCLFrameworkOptimization.h - Utility Pass for SYCL framework optimization
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+/// \file
+/// This header defines utility pass for SYCL framework optimization adding
+/// back noinline and optnone function attributes since Backends might expect
```

Comment:
Sorry I don't funny understand the intent of this pass. Is there a problem where the backend needs noinline/optnone on all functions if any functions use them otherwise it errors?

Also, I assume this happens after all optimization so it doesn't end up actually preventing any optimization of the functions we want to optimize right?

Reply:
I was asked to leave IR as "if it was from default debug pipeline".

Yes, it doesn't prevent optimizations of the functions we want to optimize.

Reply:
> I was asked to leave IR as "if it was from default debug pipeline".

Can we get a better definition of "it was from default debug pipeline"? Can we get a motivation for that? I just don't quite understand how doing partial optimization of a module co-exists with leaving IR as "if it was from default debug pipeline". To me that extra pass looks weird and it would be great to get more technical background and motivation for its existence.